WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Categories/ProductCategoryListViewModel.swift

```diff
@@ -80,6 +85,28 @@ final class ProductCategoryListViewModel {
         onSyncStateChange = onStateChanges
         onSyncStateChange?(syncCategoriesState)
     }
+
+    /// Select or Deselect a category
+    ///
+    func selectOrDeselectCategory(index: Int) {
+        guard let categoryViewModel = categoryViewModels[safe: index] else {
+            return
+        }
+
+        // If the category selected exist, remove it, otherwise, add it to `newSelectedCategories`.
+        if let indexCategory = newSelectedCategories.firstIndex(where: { $0.categoryID == categoryViewModel.categoryID}) {
+            newSelectedCategories.remove(at: indexCategory)
+        }
+        else if let category = resultController.fetchedObjects.first(where: { $0.categoryID == categoryViewModel.categoryID }) {
+            newSelectedCategories.append(category)
+        }
+
+        updateViewModelsArray()
+    }
+
+    func hasCategoriesChanges() -> Bool {
```

Comment:
nit: maybe `haveCategoriesChanged` or `hasUnsavedChanges`? it doesn't seem to be used yet?

Reply:
Thanks for the suggestion, I used `hasUnsavedChanges`, and Yes, it will be used in the next PR.